iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/schema/ttl/TTLCache.java

```diff
@@ -0,0 +1,275 @@
+    long ttl = ttlCacheTree.ttl;
+    CacheNode parent = ttlCacheTree;
+    for (int i = 1; i < nodes.length; i++) {
+      CacheNode child = parent.getChild(IoTDBConstant.MULTI_LEVEL_PATH_WILDCARD);
+      ttl = child != null ? child.ttl : ttl;
+      parent = parent.getChild(nodes[i]);
+      if (parent == null) {
+        break;
+      }
+    }
+    ttl = parent != null && parent.ttl != NULL_TTL ? parent.ttl : ttl;
+    return ttl;
+  }
+
+  public Map<String, Long> getAllTTLUnderOneNode(String[] nodes) {
+    Map<String, Long> pathTTLMap = new HashMap<>();
+    CacheNode node = ttlCacheTree;
+    for (int i = 1; i < nodes.length; i++) {
+      node = node.getChild(nodes[i]);
+      if (node == null) {
+        return pathTTLMap;
+      }
+    }
+
+    // get all ttl under current node
+    dfsCacheTree(pathTTLMap, new StringBuilder(new PartialPath(nodes).getFullPath()), node);
+    return pathTTLMap;
+  }
+
+  /**
+   * Return the ttl of path. If the path does not exist, it means that the TTL is not set, and
+   * return NULL_TTL.
+   */
+  public long getNodeTTL(String[] nodes) {
+    CacheNode node = ttlCacheTree;
+    for (int i = 1; i < nodes.length; i++) {
+      node = node.getChild(nodes[i]);
+      if (node == null) {
+        return NULL_TTL;
+      }
+    }
+    return node.ttl;
+  }
+
+  public Map<String, Long> getAllPathTTL() {
+    Map<String, Long> result = new LinkedHashMap<>();
+    dfsCacheTree(result, new StringBuilder(IoTDBConstant.PATH_ROOT), ttlCacheTree);
+    return result;
+  }
+
+  private void dfsCacheTree(Map<String, Long> pathTTLMap, StringBuilder path, CacheNode node) {
+    if (node.ttl != NULL_TTL) {
+      pathTTLMap.put(path.toString(), node.ttl);
+    }
+    int idx = path.length();
+    for (Map.Entry<String, CacheNode> entry : node.getChildren().entrySet()) {
+      dfsCacheTree(
+          pathTTLMap,
+          path.append(IoTDBConstant.PATH_SEPARATOR).append(entry.getValue().name),
+          entry.getValue());
+      path.delete(idx, path.length());
+    }
+  }
+
+  public int getTtlCount() {
+    return ttlCount;
+  }
+
+  public void serialize(BufferedOutputStream outputStream) throws IOException {
+    Map<String, Long> allPathTTLMap = getAllPathTTL();
+    ReadWriteIOUtils.write(allPathTTLMap.size(), outputStream);
+    for (Map.Entry<String, Long> entry : allPathTTLMap.entrySet()) {
+      ReadWriteIOUtils.write(entry.getKey(), outputStream);
+      ReadWriteIOUtils.write(entry.getValue(), outputStream);
+    }
+    outputStream.flush();
+  }
+
+  public void deserialize(BufferedInputStream bufferedInputStream) throws IOException {
+    int size = ReadWriteIOUtils.readInt(bufferedInputStream);
+    while (size > 0) {
+      String path = ReadWriteIOUtils.readString(bufferedInputStream);
+      long ttl = ReadWriteIOUtils.readLong(bufferedInputStream);
+      setTTL(Objects.requireNonNull(path).split("\\" + IoTDBConstant.PATH_SEPARATOR), ttl);
+      size--;
+    }
+  }
```

Comment:
Why does the parameter have to be `BufferedOutputStream` instead of the normal `OutputStream`?

Reply:
Resolved. Use `OutputStream` and `InputStream` instead.